src/dynamics/solver/joint_constraint/revolute_velocity_constraint.rs

```diff
@@ -78,9 +78,30 @@ impl RevoluteVelocityConstraint {
 
         let inv_lhs = Cholesky::new_unchecked(lhs).inverse();
 
-        let lin_rhs = rb2.linvel + rb2.angvel.gcross(r2) - rb1.linvel - rb1.angvel.gcross(r1);
-        let ang_rhs = basis1.tr_mul(&(rb2.angvel - rb1.angvel));
-        let rhs = Vector5::new(lin_rhs.x, lin_rhs.y, lin_rhs.z, ang_rhs.x, ang_rhs.y);
+        let linvel_err = rb2.linvel + rb2.angvel.gcross(r2) - rb1.linvel - rb1.angvel.gcross(r1);
+        let angvel_err = basis1.tr_mul(&(rb2.angvel - rb1.angvel));
+
+        let mut rhs = Vector5::new(
+            linvel_err.x,
+            linvel_err.y,
+            linvel_err.z,
+            angvel_err.x,
+            angvel_err.y,
+        ) * params.velocity_solve_fraction;
+
+        let velocity_based_erp_inv_dt = params.velocity_based_erp_inv_dt();
+        if velocity_based_erp_inv_dt != 0.0 {
+            let lin_err = anchor2 - anchor1;
+
+            let axis1: na::Unit<na::Vector3<_>> = rb1.position * cparams.local_axis1;
+            let axis2: na::Unit<na::Vector3<_>> = rb2.position * cparams.local_axis2;
+            let ang_err =
+                Rotation::rotation_between_axis(&axis1, &axis2).unwrap_or_else(Rotation::identity);
+            let ang_err = ang_err.scaled_axis();
+
+            rhs += Vector5::new(lin_err.x, lin_err.y, lin_err.z, ang_err.x, ang_err.y)
```

Comment:
This should also be a Vector6

Reply:
...or rather, I should use `basis1/basis2` here to get the rotation error to be two dimensional... somehow.

Reply:
narrow version fixes